portal-kernel/src/com/liferay/portal/kernel/repository/BaseRepositoryImpl.java

```diff
@@ -77,12 +77,22 @@ public FileEntry addFileEntry(
 		}
 	}
 
-	@Override
 	public abstract Folder addFolder(
 			long userId, long parentFolderId, String name, String description,
 			ServiceContext serviceContext)
 		throws PortalException;
 
+	@Override
+	public Folder addFolder(
+			String externalReferenceCode, long userId, long parentFolderId,
+			String name, String description, ServiceContext serviceContext)
+		throws PortalException {
+
+		return addFolder(
```

Comment:
I think here we are calling to the same method causing and infinite loop (stack overflow)